File: HCrawler.Core/Repositories/IImageRepository.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using HCrawler.Api.DB.Utils;
using HCrawler.Core.Repositories.Models;

namespace HCrawler.Core.Repositories
{
    public interface IImageRepository
    {
        Task<IEnumerable<DetailedImage>> GetAll(PageFilter pageFilter);
        Task<bool> ProfileExistsAsync(string profileName);
        Task<bool> SourceExistsAsync(string sourceName);
        Task<bool> ImageExistsAsync(string imagePath);
        Task<int> StoreProfileAsync(StoreProfile storeProfile);
        Task<int> StoreSourceAsync(StoreSource storeSource);
        Task<int> StoreImageAsync(StoreImage storeImage);
        Task<int> GetProfileIdByNameAsync(string profileName);
        Task<int> GetSourceIdByNameAsync(string sourceName);
    }
}

File: HCrawler.Core/IImageRepository.fs
namespace HCrawler.Core

open System.Collections.Generic
open System.Threading.Tasks
open HCrawler.Core.Payloads
open HCrawler.Core.Proxies

type IImageRepository =
    abstract getAllAsync: PageFilter -> Task<IEnumerable<DetailedImage>>
    abstract profileExistsAsync: string -> Task<bool>
    abstract sourceExistsAsync: string -> Task<bool>
    abstract imageExistsAsync: string -> Task<bool>
    abstract storeProfileAsync: StoreProfile -> Task<int>
    abstract storeSourceAsync: StoreSource -> Task<int>
    abstract storeImageAsync: StoreImage -> Task<int>
    abstract getProfileIdByNameAsync: string -> Task<int>
    abstract getSourceIdByNameAsync: string -> Task<int>